### PR TITLE
Add java docs + JUnits Test + Fixed a bug

### PR DESCRIPTION
**Pull request by joelngyx**
Added JavaDoc comments + added more JUnits Test to increase test coverage for Note, NoteList and TopicList classes + fixed a small bug

### CODE REVIEW

File: src/main/java/seedu/eduke8/note/NoteList.java [hunk 1]
```diff
@@ -17,24 +17,46 @@ public NoteList(ArrayList<Displayable> notes) {
...
+     */
     @Override
     public void add(Displayable note) {
         notes.add(note);
     }
 
+    /**
+     * Deletes a note (specified with the provided int index) from the NoteList.
+     *
+     * @param index An index, which indicates which note in the NoteList to is to be deleted.
```
Comment (anqi20) [inline src/main/java/seedu/eduke8/note/NoteList.java:33]: I think this method returns Displayable

**Review by anqi20 — CHANGES_REQUESTED** [review]
Other than that part, the rest looks good!

File: src/main/java/seedu/eduke8/note/NoteList.java (hunk 1 above)
Comment (joelngyx) [reply src/main/java/seedu/eduke8/note/NoteList.java:33]: Thanks for catching the error!

File: src/main/java/seedu/eduke8/note/NoteList.java [hunk 2]
```diff
@@ -17,24 +17,46 @@ public NoteList(ArrayList<Displayable> notes) {
...
         notes.add(note);
     }
 
+    /**
+     * Deletes a Displayable object (specified with the provided int index) from the NoteList.
+     *
+     * @param index An index, which indicates which note in the NoteList to is to be deleted.
+     */
     @Override
     public Displayable delete(int index) {
```
Comment (anqi20) [inline src/main/java/seedu/eduke8/note/NoteList.java:37]: Sorry Joel, what I meant was should there be a @return here?

Comment (joelngyx) [reply src/main/java/seedu/eduke8/note/NoteList.java:37]: Good catch! Fixed it. Thanks!

**Review by anqi20 — APPROVED** [review]
looks good!

**Pull request merged**